connect/src/main/java/com/stripe/android/connect/EmbeddedComponentManager.kt

```diff
@@ -154,42 +157,81 @@ class EmbeddedComponentManager(
             return true
         }
 
-        val launcher = getLauncher(context, requestPermissionLaunchers, "Error launching camera permission request")
-            ?: return null
+        val (_, launcher) =
+            getLauncher(context, requestPermissionLaunchers, "Error launching camera permission request")
+                ?: return null
         launcher.launch(Manifest.permission.CAMERA)
 
         return permissionsFlow.first()
     }
 
     internal suspend fun chooseFile(context: Context, requestIntent: Intent): Array<Uri>? {
-        val launcher = getLauncher(context, chooseFileLaunchers, "Error choosing file")
+        val (activity, launcher) = getLauncher(context, chooseFileLaunchers, "Error choosing file")
             ?: return null
         launcher.launch(requestIntent)
 
-        return chooseFileResultFlow.first()
+        return chooseFileResultFlow
+            .first { it.activity == activity }
+            .result
+    }
+
+    internal suspend fun presentFinancialConnections(
+        context: Context,
+        clientSecret: String,
+        connectedAccountId: String,
+    ): FinancialConnectionsSheetResult? {
+        val activity = context.findActivityWithErrorHandling()
+            ?: return null
+        val sheet = financialConnectionsSheets[activity]
+        if (sheet == null) {
+            logger.warning(
+                buildString {
+                    append("($loggerTag) Error presenting FinancialConnectionsSheet ")
+                    append("Did you call EmbeddedComponentManager.onActivityCreate in your Activity.onCreate function?")
+                }
+            )
+            return null
+        }
+        sheet.present(
+            FinancialConnectionsSheet.Configuration(
+                financialConnectionsSessionClientSecret = clientSecret,
+                publishableKey = configuration.publishableKey,
+                stripeAccountId = connectedAccountId,
+            )
+        )
+        return financialConnectionsResults
+            .first { it.activity == activity }
+            .result
```

Comment:
What's the behavior when returning `null` results here? Should we follow the approach you did in other places where we crash on debug builds?

Reply:
Yes -- pushed https://github.com/stripe/stripe-android/pull/10091/commits/3b7089cd248e31482ce56e1fe212f8361272f80d